pkg/config/config_template.yaml

```diff
@@ -1642,6 +1642,22 @@ api_key:
   #
   # trace_buffer: 0
 
+  ## @param probabilistic_sampler - object - optional
+  ## @env DD_APM_PROBABILISTIC_SAMPLER_ENABLED - boolean - optional - default: false
+  ## @env DD_APM_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE - float - optional - default: 0
+  ## @env DD_APM_PROBABILISTIC_SAMPLER_HASH_SEED - integer - optional - default: 0
+  ## Enables and configures the Probabilistic Sampler, compatible with the
+  ## OTel Probabilistic Sampler Processor ( https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/probabilisticsamplerprocessor#probabilistic-sampling-processor )
+  ## enabled: enables or disables the probabilistic sampler
+  ## sampling_percentage: samples this percentage (0-100) of traffic
+  ## hash_seed: A seed used for the hash algorithm. This must match other agents and OTel
+  ##            collectors using the probabilistic sampler to ensure consistent sampling.
+  #
+  # probabilistic_sampler:
+  #     enabled: true
+  #     sampling_percentage: 20
+  #     hash_seed: 22
```

Comment:
I think each field should be documented separately, it is at least how it's been done for similar config structures in this file.

I'll let the docs and shared component teams confirm.

Reply:
I followed the example of the filter tags. ( https://github.com/DataDog/datadog-agent/pull/23943/files/745b327df2574f458a8b50dbd659fcdbfc524f36#diff-a77843a3a2d99bdc21000210a1c6a0fc99871774b33d15b6779389e8299aa794R1495-R1507 )

Since the `probabilistic_sampler` is an object, I think they need to be documented together. Do we have a counterexample?

Alternatively, we could make each config independent and separate them here. WDYT?

Reply:
One example is the obfuscation object https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml#L1381

I don't have a strong opinion, I was mainly worried it would break any automation for the docs team bc iirc they use this file somewhere.

Reply:
Ahh, I see. Yes, I like that style better. I'll adopt it for this doc.